### DISCUSSION TITLE
How to get Helm `registryUrl` in my `postUpgradeTasks`?

### DISCUSSION THREAD

**Discussion by vmignot** [renovatebot/renovate#30671] [Request Help]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Gitlab - renovate 38.21.2

### Please tell us more about your question or problem

I am trying to run a `postUpgradeTasks` to get the default values from a Helm dependency.
In this `postUpgradeTasks`, I would like to do something like this:
```json
"postUpgradeTasks": {
  "fileFilters": ["Chart.yaml"],
  "executionMode": "update",
  "commands": [
    "helm show values --repo {{{ ??? }}} --version {{{currentVersion}}} {{{depName}}}"
  ]
}
```

What I am missing here (in the **???** placeholder) is the repo, which is displayed as `registryUrl` in my logs:
```
"deps": [
               {
                 "depName": "gitlab",
                 "currentValue": "8.2.1",
                 "registryUrls": ["https://charts.gitlab.io/"],
                 "updates": [
                   {
                     "bucket": "non-major",
                     "newVersion": "8.2.2",
                     "newValue": "8.2.2",
                     "newDigest": "63feeb33c9cc9e9ac044a06738762e4c3f75f4eced7c3a0ededbd361e5bc6be2",
                     "releaseTimestamp": "2024-08-07T08:50:13.397Z",
                     "newMajor": 8,
                     "newMinor": 2,
                     "newPatch": 2,
                     "updateType": "patch",
                     "branchName": "renovate/main-gitlab-8.x"
                   }
                 ],
                 "packageName": "gitlab",
                 "versioning": "helm",
                 "warnings": [],
                 "sourceUrl": "https://gitlab.com/gitlab-org/charts/gitlab",
                 "registryUrl": "https://charts.gitlab.io/",
                 "homepage": "https://about.gitlab.com/",
                 "currentVersion": "8.2.1",
                 "currentVersionTimestamp": "2024-07-24T13:48:46.402Z",
                 "isSingleVersion": true,
                 "fixedVersion": "8.2.1"
               },
```

I have tried to find it in the [https://docs.renovatebot.com/templates](https://docs.renovatebot.com/templates) documentation page, without success...

Do you have an idea on how to achieve this?
Thanks!

**Comment by RahulGautamSingh** [comment]
You could try the `releases` field. 
Example:  ``` helm show values --repo {{{ releases[0].regsitryUrl }}} --version {{{currentVersion}}} {{{depName}}} ```

**Reply by vmignot** [reply]
Hello, thanks for your time :-)

I tried, and got the following problem:

> ### ⚠️ Artifact update problem
> 
> Renovate failed to update an artifact related to this branch. You probably do not want to merge this MR as-is.
> 
> ♻ Renovate will retry this branch, including artifacts, only when one of the following happens:
> 
>  - any of the package files in this branch needs updating, or 
>  - the branch becomes conflicted, or
>  - you click the rebase/retry checkbox if found above, or
>  - you rename this MR's title to start with "rebase!" to trigger it manually
> 
> The artifact failure details are included below:
> 
> ##### File name: infra/gitlab/Chart.yaml
> 
> ```
> Parse error on line 1:
> ...ow values --repo {{{releases[0].regsitry
> -----------------------^
> Expecting 'ID', 'STRING', 'NUMBER', 'BOOLEAN', 'UNDEFINED', 'NULL', 'DATA', got 'INVALID'
> ```

In the documentation, it says that: "`logJSON` and `releases` are only allowed in commitBody template.". Does this mean that I cannot use it in my case?

**Reply by RahulGautamSingh** [reply]
Sorry, my bad. I missed that line while reading. I don't think it is possible then to get the result you want.

**Reply by vmignot** [reply]
Thanks, it's sad :-(
I really do want to run a postUpgradeTasks where I can see what has changed in the default chart values...

Is this something that Renovate would consider exposing if I submit a PR or is there another reason why it is not exposed to the template?

**Reply by RahulGautamSingh** [reply]
Lets wait for a maintainers comment I will tag them

**Reply by JarneVdP** [reply]
I don't think/know if this will solve your problem but you have a typo in releases[0].**regsitry**Url. If that is the correct value, then it probably should be  releases[0].registryUrl. From experience you might have to use releases.registryUrl however even that might not work.

**Reply by vmignot** [reply]
Hello, no sorry that was not it.
It was still worth trying :-)

**Comment by ChristianCiach** [comment]
Same issue, different use-case: I would like to call `helm template` as part of `postUpgradeTasks` to commit the rendered YAML of the chart. I tried `{{{registryUrl}}}` as a variable, but to no avail.
